src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/BStrStringMarshaller.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Runtime.InteropServices.Marshalling
+{
+    /// <summary>
+    /// Marshaller for BSTR strings
+    /// </summary>
+    [CLSCompliant(false)]
+    [CustomTypeMarshaller(typeof(string), BufferSize = 0x200,
```

Comment:
```suggestion
    [CustomTypeMarshaller(typeof(string), BufferSize = 0x100,
```
The size is no ushort's now